lib/gradebook.rb

```diff
@@ -1 +1,52 @@
+class Gradebook
+  attr_reader :instructor, :courses, :all_students
+  def initialize (instructor)
+    @instructor = instructor
+    @courses = []
+    @all_students_a
+  end
 
+  def add_course(course)
+    @courses << course
+  end
+
+  def all_students
+    all_students = []
+    @courses.each {|course| all_students << course.students}
+    all_students.flatten
+  end
+
+  def list_all_students
+    all_students_h = Hash.new {|hash, key| hash[key] = []}
+    @courses.each do |course|
+      all_students_h[course.name] = course.students
```

Comment:
According to the interaction pattern, the key for this hash should be the whole `course` instance, not just the name attribute.
```suggestion
      all_students_h[course] = course.students
```